File: bdd-java/src/test/java/steps/Steps.java
package steps;

import bdd.WordFrequencyReporter;
import cucumber.api.java8.En;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Steps implements En {
    public Steps() {

        String wordList = "";
        HashMap<String, String> state = new HashMap<>();

        Given("a list of words (.*)", (String list) -> {
            state.put("list", list);
        });

        And("the number of top repeated words to print is (\\d+)", (Integer count) -> {
            state.put("count", "" + count);
        });

        When("the word frequency program is executed", () -> {
            List<String> list = new ArrayList<String>(Arrays.asList(state.get("list").split(",")));
            Integer count = Integer.parseInt(state.get("count"));
            state.put("results", new WordFrequencyReporter().calculate(list));
        });

        Then("the top (\\d+) most frequently occurring words and frequency are calculated", (Integer count) -> {
            System.out.println("Count : " + count);
        });

        And("the results should be (.*)", (String output) -> {
            assertEquals("should be equal", state.get("results"), output);
        });
    }
}

File: bdd-python/features/steps/steps.py

from behave import given, when, then
from WordFrequencyReporter import WordFrequencyReporter

@given('a list of words {listOfWords}')
def given_a_list_of_words(context, listOfWords):
    print("List : " + listOfWords)
    context.listOfWords = listOfWords
 
@given('the number of top repeated words to print is {count:d}')
def the_number_is(context, count):
    context.countSummary = count

@when('the word frequency program is executed') 
def the_world_frequency_program_is_executed(context):
    context.actualResults = WordFrequencyReporter().calculate()

@then('the top {count:d} most frequently occurring words and frequency are printed')
def check_is_printed(context, count):
    print("Count : " + count)

@then('the results should be {output}') 
def check_results(context, output):
    assert context.actualResults is output
